src/kv/set.h

```diff
@@ -36,9 +36,49 @@ namespace kv
     using WriteOnlyHandle = kv::WriteableSetHandle<K, KSerialiser, Unit>;
     using Handle = kv::SetHandle<K, KSerialiser, Unit>;
 
-    using KeySerialiser = KSerialiser;
+    // Note: The type V of the value `std::optional<V>` does not matter here.
+    // The optional type is required to differentiate additions from deletions,
+    // and to provide a consistent interface with the more generic `TypedMap`.
+    using Write = std::map<K, std::optional<kv::serialisers::SerialisedEntry>>;
```

Comment:
The hook for `kv::Set` is a little awkward because we need to be able to differentiate additions from deletions (of a key). We could pass two separate lists to the hooks, but it would be inconsistent with the hook on `TypedMap`. Instead, we use a single map, with an optional value whose type `kv::serialisers::SerialisedEntry` doesn't really matter.